src/js/actions/tools.js

```diff
@@ -790,6 +808,37 @@ define(function (require, exports) {
             modifiers: {},
             fn: _vectorMaskHandler
         });
+        var DEBOUNCE_DELAY = 200;
+
+        // recreate pointer policies if we're in Vector Mask Mode on UI Store change events
+        _vectorMaskPolicyHandler = synchronization.debounce(function () {
+            var vectorMaskMode = toolStore.getVectorMode();
+
+            if (vectorMaskMode) {
+                var pointerPolicyID = toolStore.getVectorMaskPolicyID();
```

Comment:
I think it would be preferable for the entire function to be an action, which then performs these transfers. What should be debounced is the call to the `handleVectorMaskModeChange` action.